src/parsing/binast-parse-data.cc

```diff
@@ -60,16 +76,45 @@ Handle<BinAstParseData> ZoneProducedBinAstParseData::Serialize(OffThreadIsolate*
 
 Handle<BinAstParseData> ZoneBinAstParseData::Serialize(Isolate* isolate) {
   int data_size = static_cast<int>(byte_data()->size());
-  Handle<ByteArray> serialized_ast = isolate->factory()->NewByteArray(data_size, AllocationType::kOld);
+  Handle<ByteArray> serialized_ast =
+      isolate->factory()->NewByteArray(data_size, AllocationType::kOld);
   serialized_ast->copy_in(0, byte_data()->data(), data_size);
-  return isolate->factory()->NewBinAstParseData(serialized_ast);
+
+  int offsets_map_size = static_cast<int>(function_literal_offsets_.size());
+  Handle<PodArray<BinAstInnerFunctionLiteralOffset>> inner_function_offsets =
+      PodArray<BinAstInnerFunctionLiteralOffset>::New(isolate, offsets_map_size,
+                                                      AllocationType::kOld);
+
+  int index = 0; 
+  for (auto offset : function_literal_offsets_) {
+    inner_function_offsets->set(index++, offset);
+  }
+
+  return isolate->factory()->NewBinAstParseData(serialized_ast,
+                                                inner_function_offsets);
 }
 
-Handle<BinAstParseData> ZoneBinAstParseData::Serialize(OffThreadIsolate* isolate) {
+Handle<BinAstParseData> ZoneBinAstParseData::Serialize(
+    OffThreadIsolate* isolate) {
   int data_size = static_cast<int>(byte_data()->size());
-  Handle<ByteArray> serialized_ast = isolate->factory()->NewByteArray(data_size, AllocationType::kOld);
+  Handle<ByteArray> serialized_ast =
+      isolate->factory()->NewByteArray(data_size, AllocationType::kOld);
   serialized_ast->copy_in(0, byte_data()->data(), data_size);
-  return isolate->factory()->NewBinAstParseData(serialized_ast);
+
+  int offsets_map_size = static_cast<int>(function_literal_offsets_.size());
+
+  Handle<PodArray<BinAstInnerFunctionLiteralOffset>> inner_function_offsets =
+      Handle<PodArray<BinAstInnerFunctionLiteralOffset>>::cast(
+          isolate->factory()->NewByteArray(offsets_map_size,
+                                           AllocationType::kOld));
+
+  int index = 0;  // Index into the {PodArray} above.
+  for (auto offset : function_literal_offsets_) {
+    inner_function_offsets->set(index++, std::move(offset));
```

Comment:
No need to `std::move` here, we don't need move semantics since the rvalue is an instance of a POD class.